### ISSUE TITLE
feature: Require password for sensitive transactions: Password change and Token creation

### ISSUE THREAD

**Issue by richard-stafflink** [windmill-labs/windmill#2516]
The user's password should be required for sensitive transaction like changing their password and generating tokens.

https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html#require-re-authentication-for-sensitive-features

**Comment by rubenfiszel** [comment]
Agreed. What is the recommendation when the user has no password (using oauth for instance) ? Re-login again with OAuth ?

**Comment by richard-stafflink** [comment]
Good question.

I set up a new account in Gitlab with just OAuth, and I was able to add new SSH and GPG keys without being prompted for any other auth type. There was also the option to set a new password, without further verification.

2FA is an option to protect a user for the OAuth case - but turning on 2FA is voluntary.
A form of 2FA would be a code (or an "Authorise" link) sent to their email - so that is an option for OAuth users.

I guess if a password has been set, verifying the user's password should occur, but with the case of OAuth being the only authentication method enabled, then you can only suggest they enable 2FA - Then the onus is back on the user.

Most organisations these days mandate all their employees enable 2FA, which has been the case for all the companies I have worked for in the last 10 years.
